sbids/tasks/lotkavolterra.py

```diff
@@ -0,0 +1,187 @@
+import jax
+from jax import lax
+import jax.numpy as jnp
+from jax.experimental.ode import odeint
+from tensorflow_probability.substrates import jax as tfp
+tfd = tfp.distributions
+tfb = tfp.bijectors
+
+
+class Lotka_Volterra:
+
+    def __init__(self,time=10,nb_statistics=10):
+        """Lotka Voleterra 
+        Parameters
+        ----------
+        time: Number of measurement times.
+        nb_statistics: Number of statistics for the observation vector. It must be divisible by 2.
+        """
+        self.time = time
+        self.nb_statistics = nb_statistics
+
+
+
+        def equation(y,t, theta):
+            """
+            Evaluate the Lotka-Volterra time derivative of the solution `y` at time
+            `t` as `func(y, t, *args)` to feed odeint.
+            Parameters
+            ----------
+            y: float
+            The two populations.
+            t : float
+            The time.
+            theta: float
+            Model parameters.
+            Returns
+            -------
+            Lotka-Volterra equations.
+            """
+            X = y[0]
+            Y = y[1]
+
+            alpha, beta, gamma, delta = (
+                theta[..., 0],
+                theta[..., 1],
+                theta[..., 2],
+                theta[..., 3],)
+
+            dX_dt = alpha * X - beta * X * Y
+            dY_dt = -gamma * Y + delta * X * Y
+            return jnp.stack([dX_dt, dY_dt],axis=0)
+        self.equation = equation
+
+        @jax.jit
+        def solve_equation(init,theta):
+            nb_statistics = self.nb_statistics / 2 
+            step = self.time / nb_statistics
+            ts = jnp.arange(0.,time, step)
+            z = odeint(equation, init, ts, theta, rtol=1e-6, atol=1e-5)
+            z = z.T
+
+            return z
+        self.solve_equation = solve_equation
+
+
+    def simulator(self, seed, batch_size=100, which_score=0):
+        """
+        Generate dataset (theta, observation, score) with stochastic initial 
+        conditions sampled from LogNormal(log(10),0.8).
+        Parameters
+        ----------
+        key: PRNGKeyArray
+        batch_size: int
+            Size of the batch.
+        which_score: int 
+            0 -> Grad_theta(log p(theta|x,z))
+            1 -> Grad_theta(log p(x,z|theta))
+        Returns
+        -------
+            Dataset (theta, observation, score).
+            'Theta': the model parameters.
+            'Observation': the simulations from p(x|theta). 
+            'Score': the joint score.
+        """
+
+        key1, key2, key3, key4 = jax.random.split(seed,4)
+
+        prior_parameters = tfd.Independent(tfd.LogNormal(jnp.array([-0.125,-3,-0.125,-3]), 0.5*jnp.ones(4)),1)
+        prior_latent_variable = tfd.LogNormal(jnp.log(10*jnp.ones(2)), 0.8*jnp.ones(2))
```

Comment:
case and point of why it's very bad practice to duplicate code >.< the prior on latents here is different from the one in the `get_reference_posterior` function which is:
```
tfd.LogNormal(jnp.log(jnp.ones(2)*3), 0.5*jnp.ones(2))
```